octobot_trading/exchanges/connectors/ccxt_exchange.py

```diff
@@ -271,125 +431,695 @@ async def get_order_book(self, symbol: str, limit: int = 5, **kwargs: dict) -> t
+                               ) -> dict or typing.Tuple[dict, dict]:
         try:
             with self.error_describer():
-                return await self.client.fetch_ticker(symbol, params=kwargs)
+                raw_ticker = await self.client.fetch_ticker(symbol, params=kwargs)
+            return await self.parse_ticker(
+                raw_ticker=raw_ticker,
+                symbol=symbol, also_get_mini_ticker=also_get_mini_ticker)
         except ccxt.NotSupported:
             raise octobot_trading.errors.NotSupported
         except ccxt.BaseError as e:
             raise octobot_trading.errors.FailedRequest(f"Failed to get_price_ticker {e}")
 
-    async def get_all_currencies_price_ticker(self, **kwargs: dict) -> typing.Optional[list]:
+    async def get_all_currencies_price_ticker(self, **kwargs: dict) -> list:
         try:
             with self.error_describer():
                 symbols = kwargs.pop("symbols", None)
-                self.all_currencies_price_ticker = await self.client.fetch_tickers(symbols, params=kwargs)
+                self.all_currencies_price_ticker = await self.parse_tickers(
+                    await self.client.fetch_tickers(symbols, params=kwargs))
             return self.all_currencies_price_ticker
         except ccxt.NotSupported:
             raise octobot_trading.errors.NotSupported
         except ccxt.BaseError as e:
             raise octobot_trading.errors.FailedRequest(f"Failed to get_all_currencies_price_ticker {e}")
 
-    # ORDERS
-    async def get_order(self, order_id: str, symbol: str = None, **kwargs: dict) -> dict:
-        if self.client.has['fetchOrder']:
+    async def get_order(self, order_id: str, symbol: str = None, check_completeness: bool = True,
+                        **kwargs: dict) -> typing.Optional[dict]:
+        defined_methods = self.connector_config.GET_ORDER_METHODS
+        if self.get_order_default.__name__ in defined_methods and \
+                (order := await self.get_order_default(order_id, symbol,
+                                                       check_completeness=check_completeness, **kwargs)):
+            return order
+        if self.get_order_from_open_and_closed_orders.__name__ in defined_methods and \
+                (order := await self.get_order_from_open_and_closed_orders(order_id, symbol,
+                                                                           check_completeness=check_completeness,
+                                                                           **kwargs)):
+            return order
+        if self.get_order_using_stop_params.__name__ in defined_methods and \
+                (order := await self.get_order_using_stop_params(order_id, symbol,
+                                                                 check_completeness=check_completeness)):
+            return order
+        if self.get_trade.__name__ in defined_methods and \
```

Comment:
Why do we found a get_trade call in a get_order method ? I think those edge cases have to be handled in an exchange tentacle

Reply:
I change the default connector settings. Its still there, but will not be called with default config

Reply:
My question is: why calling `get_trade` in `get_order`?

Reply:
The second point is: I think we should let the tentacle handle the edge cases of the exchange like calling `get_order_from_open_and_closed_orders` if needed instead of just `get_order_default`.

Reply:
there was a method called get_order_from_trades in spot_ccxt_exchange which i just renamed to get_trade and moved to here.
We can have it here as a fallback as a parsed trade is basically just a closed order.
Some exchanges dont provide closed orders, in those cases it might help

get_order_from_open_and_closed_orders and get_trade is now removed from the default settings, but I would suggest keeping it as a default as it will never be called by an exchange that works properly.
Only exchanges that would fail otherwise have a chance of success

What I'll do is make two different CCXTExchangeConfigs:
1. the default one will be as generic as possible (without causing potential harm)
2. the other one has only settings and parser logic according to ccxt standards

the process of adding support for an exchange would be to set the Config Class to ccxt standards and apply patches ontop of it. That way tested exchanges will never use generic stuff

Reply:
Concerning the get_trade I understand that we can fetch some orders in specific cases with it. 
My point is that we shouldn't use this fallback methode by default. I think we need to use it on specific exchange (so the config class with patches you explained in the end).

Reply:
I totally agree with a default ccxt parser that we can override to fix some specific stuff

Reply:
To resume my point of view : we should have those methods (get_order_from_open_and_closed_orders, get_trade, ...) available but not called in the default implementation. We should only call them when we know it's the correct way to get the order

Reply:
We now have two different ccxt settings one is according to ccxt standards and the other generic ccxt.
Exchanges without official support use the generic one and supported ones use according to ccxt standards.
I would say we discuss the default settings for the generic one once we want to merge it as, its just a switch anyway